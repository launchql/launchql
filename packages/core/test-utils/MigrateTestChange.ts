export interface MigrateTestChange {
  name: string;
  dependencies?: string[];
  timestamp?: string;
  planner?: string;
  email?: string;
  comment?: string;
}
