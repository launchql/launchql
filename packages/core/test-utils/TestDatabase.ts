import { PgConfig } from 'pg-env';

export interface TestDatabase {
  name: string;
  config: PgConfig;
  query(sql: string, params?: any[]): Promise<any>;
  exists(type: 'schema' | 'table', name: string): Promise<boolean>;
  getDeployedChanges(): Promise<any[]>;
  getDependencies(packageName: string, changeName: string): Promise<string[]>;
  getMigrationState(): Promise<{
    changes: any[];
    events: any[];
    changeCount: number;
    eventCount: number;
  }>;
  close(): Promise<void>;
}
