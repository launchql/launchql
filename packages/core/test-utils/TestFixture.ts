import fs from 'fs';
import os from 'os';
import path from 'path';

import { LaunchQLPackage } from '../src';
import { getFixturePath } from './utils';

const { mkdtempSync, rmSync, cpSync } = fs;

export class TestFixture {
  readonly tempDir: string;
  readonly tempFixtureDir: string;
  readonly getFixturePath: (...paths: string[]) => string;
  readonly getModuleProject: (workspacePath: string[], moduleName: string) => LaunchQLPackage;
  
  constructor(...fixturePath: string[]) {
    const originalFixtureDir = getFixturePath(...fixturePath);
    this.tempDir = mkdtempSync(path.join(os.tmpdir(), 'launchql-test-'));
    this.tempFixtureDir = path.join(this.tempDir, ...fixturePath);
  
    cpSync(originalFixtureDir, this.tempFixtureDir, { recursive: true });
  
    this.getFixturePath = (...paths: string[]) =>
      path.join(this.tempFixtureDir, ...paths);
  
    this.getModuleProject = (workspacePath: string[], moduleName: string): LaunchQLPackage => {
      const workspace = new LaunchQLPackage(this.getFixturePath(...workspacePath));
      const moduleMap = workspace.getModuleMap();
      const meta = moduleMap[moduleName];
      if (!meta) throw new Error(`Module ${moduleName} not found in workspace`);
      return new LaunchQLPackage(this.getFixturePath(...workspacePath, meta.path));
    };
  }
  
  fixturePath(...paths: string[]) {
    return path.join(this.tempFixtureDir, ...paths);
  }
  
  cleanup() {
    rmSync(this.tempDir, { recursive: true, force: true });
  }
}
  