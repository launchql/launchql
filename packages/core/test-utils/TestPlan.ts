import { existsSync } from 'fs';
import { dirname,join } from 'path';

import { parsePlanFile } from '../src/files';
import { ExtendedPlanFile, ParseResult } from '../src/files/types';

export class TestPlan {
  private fixturePath: string;
  result: ParseResult<ExtendedPlanFile>;

  constructor(fixturePath: string) {
    // Otherwise, resolve relative to root __fixtures__/sqitch-plans
    const rootDir = dirname(dirname(dirname(__dirname))); // Go up to workspace root
    const basePath = join(rootDir, '__fixtures__', 'sqitch-plans');
    this.fixturePath = join(basePath, fixturePath);
    
    if (!existsSync(this.fixturePath)) {
      throw new Error(`Fixture not found: ${this.fixturePath}`);
    }

    this.result = parsePlanFile(this.fixturePath);
  }

  expectResult(expected: ParseResult<ExtendedPlanFile>) {
    expect(this.result.errors).toEqual(expected.errors);
    expect(this.result.data).toEqual(expected.data);
  }
}