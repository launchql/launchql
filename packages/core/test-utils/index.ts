export { TestPlan } from './TestPlan';
export { CoreDeployTestFixture } from './CoreDeployTestFixture';
export { TestFixture } from './TestFixture';

export { MigrateTestFixture } from './MigrateTestFixture';
export { TestDatabase } from './TestDatabase';
export { MigrateTestChange } from './MigrateTestChange';

export {
  FIXTURES_PATH,
  getFixturePath,
  cleanText,
  teardownAllPools,
  closeDatabasePools
} from './utils';
