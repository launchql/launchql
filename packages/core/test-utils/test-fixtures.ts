import { LaunchQL } from '../src/class/launchql-refactored';
import fs from 'fs';
import os from 'os';
import path from 'path';

const { mkdtempSync, rmSync, cpSync } = fs;

export const FIXTURES_PATH = path.resolve(__dirname, '../../../__fixtures__');

export const getFixturePath = (...paths: string[]) =>
  path.join(FIXTURES_PATH, ...paths);

export const cleanText = (text: string): string =>
  text
    .split('\n')
    .map((line) => line.trim())
    .filter((line) => line.length > 0)
    .join('\n');

/**
 * Test fixture for the refactored LaunchQL class
 * Provides utilities for creating temporary test environments
 */
export class LaunchQLTestFixture {
  readonly tempDir: string;
  readonly tempFixtureDir: string;
  readonly getFixturePath: (...paths: string[]) => string;
  readonly getLaunchQL: (workspacePath?: string[]) => LaunchQL;
  readonly getModuleLaunchQL: (workspacePath: string[], moduleName: string) => LaunchQL;

  constructor(...fixturePath: string[]) {
    const originalFixtureDir = getFixturePath(...fixturePath);
    this.tempDir = mkdtempSync(path.join(os.tmpdir(), 'launchql-test-'));
    this.tempFixtureDir = path.join(this.tempDir, ...fixturePath);

    cpSync(originalFixtureDir, this.tempFixtureDir, { recursive: true });

    this.getFixturePath = (...paths: string[]) =>
      path.join(this.tempFixtureDir, ...paths);

    // Create LaunchQL instance for workspace or specific path
    this.getLaunchQL = (workspacePath?: string[]): LaunchQL => {
      const cwd = workspacePath 
        ? this.getFixturePath(...workspacePath)
        : this.tempFixtureDir;
      return new LaunchQL(cwd);
    };

    // Create LaunchQL instance for a specific module
    this.getModuleLaunchQL = (workspacePath: string[], moduleName: string): LaunchQL => {
      const workspace = new LaunchQL(this.getFixturePath(...workspacePath));
      const moduleMap = workspace.getModuleMap();
      const meta = moduleMap[moduleName];
      if (!meta) throw new Error(`Module ${moduleName} not found in workspace`);
      return new LaunchQL(this.getFixturePath(...workspacePath, meta.path));
    };
  }

  /**
   * Get the full path to a file/directory within the fixture
   */
  fixturePath(...paths: string[]) {
    return path.join(this.tempFixtureDir, ...paths);
  }

  /**
   * Create a new file in the fixture
   */
  createFile(relativePath: string, content: string) {
    const fullPath = this.fixturePath(relativePath);
    const dir = path.dirname(fullPath);
    if (!fs.existsSync(dir)) {
      fs.mkdirSync(dir, { recursive: true });
    }
    fs.writeFileSync(fullPath, content);
  }

  /**
   * Read a file from the fixture
   */
  readFile(relativePath: string): string {
    return fs.readFileSync(this.fixturePath(relativePath), 'utf-8');
  }

  /**
   * Check if a file exists in the fixture
   */
  fileExists(relativePath: string): boolean {
    return fs.existsSync(this.fixturePath(relativePath));
  }

  /**
   * List files in a directory
   */
  listFiles(relativePath: string = ''): string[] {
    const dir = this.fixturePath(relativePath);
    if (!fs.existsSync(dir) || !fs.statSync(dir).isDirectory()) {
      return [];
    }
    return fs.readdirSync(dir);
  }

  /**
   * Clean up the temporary directory
   */
  cleanup() {
    rmSync(this.tempDir, { recursive: true, force: true });
  }
}

/**
 * Helper to create a minimal workspace fixture
 */
export function createMinimalWorkspace(): LaunchQLTestFixture {
  const fixture = new LaunchQLTestFixture();
  
  // Create minimal launchql.json
  fixture.createFile('launchql.json', JSON.stringify({
    modules: {},
    extensions: {}
  }, null, 2));

  return fixture;
}

/**
 * Helper to create a workspace with a module
 */
export function createWorkspaceWithModule(moduleName: string, modulePath: string = 'packages/' + moduleName): LaunchQLTestFixture {
  const fixture = new LaunchQLTestFixture();
  
  // Create workspace config
  fixture.createFile('launchql.json', JSON.stringify({
    modules: {
      [moduleName]: {
        path: modulePath
      }
    },
    extensions: {}
  }, null, 2));

  // Create module directory
  fixture.createFile(path.join(modulePath, '.gitkeep'), '');

  // Create module sqitch.conf
  fixture.createFile(path.join(modulePath, 'sqitch.conf'), `[core]
	engine = pg
	plan_file = sqitch.plan
[engine "pg"]
	target = db:pg:
[deploy]
	verify = true
[rebase]
	verify = true
`);

  // Create empty plan
  fixture.createFile(path.join(modulePath, 'sqitch.plan'), `%syntax-version=1.0.0
%project=${moduleName}

`);

  // Create deploy/revert/verify directories
  fixture.createFile(path.join(modulePath, 'deploy', '.gitkeep'), '');
  fixture.createFile(path.join(modulePath, 'revert', '.gitkeep'), '');
  fixture.createFile(path.join(modulePath, 'verify', '.gitkeep'), '');

  return fixture;
}

/**
 * Helper to add a change to a module
 */
export function addChangeToModule(
  fixture: LaunchQLTestFixture, 
  modulePath: string,
  changeName: string,
  dependencies: string[] = []
): void {
  const deployPath = path.join(modulePath, 'deploy', `${changeName}.sql`);
  const revertPath = path.join(modulePath, 'revert', `${changeName}.sql`);
  const verifyPath = path.join(modulePath, 'verify', `${changeName}.sql`);

  // Create SQL files
  fixture.createFile(deployPath, `-- Deploy ${changeName}\nBEGIN;\n\n-- Your deploy SQL here\n\nCOMMIT;\n`);
  fixture.createFile(revertPath, `-- Revert ${changeName}\nBEGIN;\n\n-- Your revert SQL here\n\nCOMMIT;\n`);
  fixture.createFile(verifyPath, `-- Verify ${changeName}\nBEGIN;\n\n-- Your verify SQL here\n\nROLLBACK;\n`);

  // Update plan file
  const planPath = path.join(modulePath, 'sqitch.plan');
  const currentPlan = fixture.readFile(planPath);
  const timestamp = new Date().toISOString().replace(/\.\d{3}Z$/, 'Z');
  const deps = dependencies.length > 0 ? `[${dependencies.join(' ')}] ` : '';
  const newLine = `${changeName} ${deps}${timestamp} Test User <test@example.com> # Add ${changeName}\n`;
  
  fixture.createFile(planPath, currentPlan + newLine);
}

/**
 * Helper to create a test database configuration
 */
export function createTestDatabaseConfig(fixture: LaunchQLTestFixture, modulePath: string): void {
  const sqitchConf = `[core]
	engine = pg
	plan_file = sqitch.plan
[engine "pg"]
	target = db:pg://postgres@localhost/testdb
[target "test"]
	uri = db:pg://postgres@localhost/testdb
[deploy]
	verify = true
[rebase]
	verify = true
`;
  
  fixture.createFile(path.join(modulePath, 'sqitch.conf'), sqitchConf);
}