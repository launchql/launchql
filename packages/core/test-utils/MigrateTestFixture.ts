import { cpSync, mkdirSync,mkdtempSync, rmSync, writeFileSync } from 'fs';
import { tmpdir } from 'os';
import { join } from 'path';
import { Pool } from 'pg';
import { getPgPool } from 'pg-cache';
import { PgConfig } from 'pg-env';
import { getPgEnvOptions } from 'pg-env';

import { LaunchQLMigrate } from '../src/migrate/client';
import { MigrateTestChange } from './MigrateTestChange';
import { TestDatabase } from './TestDatabase';
import { FIXTURES_PATH } from './utils';

export class MigrateTestFixture {
  private tempDirs: string[] = [];
  private databases: TestDatabase[] = [];
  private dbCounter = 0;
  private pools: Pool[] = [];

  async setupTestDatabase(): Promise<TestDatabase> {
    const dbName = `test_migrate_${Date.now()}_${Math.random().toString(36).substring(2, 8)}_${this.dbCounter++}`;
    
    // Get base config from environment using pg-env
    const baseConfig = getPgEnvOptions({
      database: 'postgres'
    });
    
    // Create database using admin pool
    const adminPool = getPgPool(baseConfig);
    try {
      await adminPool.query(`CREATE DATABASE "${dbName}"`);
    } catch (e) {
      if (e && typeof e === 'object' && 'errors' in e && Array.isArray((e as any).errors)) {
        for (const err of (e as any).errors) {
          console.error('AggregateError item:', err);
        }
      } else {
        console.error('Test failure:', e);
      }
      throw e;
    }
    
    // Get config for the new test database
    const pgConfig = getPgEnvOptions({
      database: dbName
    });
    
    const config: PgConfig = {
      host: pgConfig.host,
      port: pgConfig.port,
      user: pgConfig.user,
      password: pgConfig.password,
      database: pgConfig.database
    };

    // Initialize migrate schema
    const migrate = new LaunchQLMigrate(config);
    await migrate.initialize();

    // Get pool for test database operations
    const pool = getPgPool(pgConfig);
    this.pools.push(pool);

    const fixture = this;
    const db: TestDatabase = {
      name: dbName,
      config,
      
      async query(sql: string, params?: any[]) {
        return pool.query(sql, params);
      },

      async exists(type: 'schema' | 'table', name: string) {
        if (type === 'schema') {
          const result = await pool.query(
            `SELECT EXISTS (
              SELECT 1 FROM information_schema.schemata 
              WHERE schema_name = $1
            ) as exists`,
            [name]
          );
          return result.rows[0].exists;
        } else {
          const [schema, table] = name.includes('.') ? name.split('.') : ['public', name];
          const result = await pool.query(
            `SELECT EXISTS (
              SELECT 1 FROM information_schema.tables 
              WHERE table_schema = $1 AND table_name = $2
            ) as exists`,
            [schema, table]
          );
          return result.rows[0].exists;
        }
      },

      async getDeployedChanges() {
        const result = await pool.query(
          `SELECT project, change_name, deployed_at 
           FROM launchql_migrate.changes 
           ORDER BY deployed_at`
        );
        return result.rows;
      },

      async getMigrationState() {
        const changes = await pool.query(`
          SELECT project, change_name, script_hash, deployed_at
          FROM launchql_migrate.changes 
          ORDER BY deployed_at
        `);
        
        const events = await pool.query(`
          SELECT project, change_name, event_type, occurred_at, error_message, error_code
          FROM launchql_migrate.events 
          ORDER BY occurred_at
        `);
        
        const sanitizedEvents = events.rows;
        
        // Remove timestamps from objects for consistent snapshots
        const cleanChanges = changes.rows.map(({ deployed_at, ...change }) => change);
        const cleanEvents = sanitizedEvents.map(({ occurred_at, ...event }) => event);
        
        return {
          changes: cleanChanges,
          events: cleanEvents,
          changeCount: cleanChanges.length,
          eventCount: cleanEvents.length
        };
      },

      async getDependencies(project: string, changeName: string) {
        const result = await pool.query(
          `SELECT d.requires 
           FROM launchql_migrate.dependencies d
           JOIN launchql_migrate.changes c ON c.change_id = d.change_id
           WHERE c.project = $1 AND c.change_name = $2`,
          [project, changeName]
        );
        return result.rows.map((row: any) => row.requires);
      },

      async close() {
        // Don't close the pool here as it's managed by pg-cache
        // Just mark this database as closed
      }
    };

    this.databases.push(db);
    return db;
  }


  setupFixture(fixturePath: string[]): string {
    const originalPath = join(FIXTURES_PATH, ...fixturePath);
    const fixtureName = fixturePath[fixturePath.length - 1]; // Use last element as fixture name
    const tempDir = mkdtempSync(join(tmpdir(), 'migrate-test-'));
    const fixtureDestPath = join(tempDir, fixtureName);
    
    cpSync(originalPath, fixtureDestPath, { recursive: true });
    this.tempDirs.push(tempDir);
    
    return fixtureDestPath;
  }

  createPlanFile(project: string, changes: MigrateTestChange[]): string {
    const tempDir = mkdtempSync(join(tmpdir(), 'migrate-test-'));
    this.tempDirs.push(tempDir);

    const lines = [
      '%syntax-version=1.0.0',
      `%project=${project}`,
      `%uri=https://github.com/test/${project}`,
      ''
    ];

    for (const change of changes) {
      let line = change.name;
      
      if (change.dependencies && change.dependencies.length > 0) {
        line += ` [${change.dependencies.join(' ')}]`;
      }
      
      line += ` ${change.timestamp || new Date().toISOString().replace(/\.\d{3}Z$/, 'Z')}`;
      line += ` ${change.planner || 'test'}`;
      line += ` <${change.email || 'test@example.com'}>`;
      
      if (change.comment) {
        line += ` # ${change.comment}`;
      }
      
      lines.push(line);
    }

    const planPath = join(tempDir, 'launchql.plan');
    writeFileSync(planPath, lines.join('\n'));
    
    return tempDir;
  }

  createScript(dir: string, type: 'deploy' | 'revert' | 'verify', name: string, content: string): void {
    const scriptDir = join(dir, type);
    mkdirSync(scriptDir, { recursive: true });
    writeFileSync(join(scriptDir, `${name}.sql`), content);
  }

  async cleanup(): Promise<void> {
    // Close all test database pools FIRST
    for (const pool of this.pools) {
      try {
        await pool.end();
      } catch (e) {
        // Ignore errors during pool closure
      }
    }
    
    // Clear the pools array
    this.pools = [];

    // Remove temporary directories
    for (const dir of this.tempDirs) {
      try {
        rmSync(dir, { recursive: true, force: true });
      } catch (e) {
        // Ignore errors during cleanup
      }
    }
    
    // Clear the databases array
    this.databases = [];
  }
}
