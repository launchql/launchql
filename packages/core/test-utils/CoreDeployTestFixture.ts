import { getEnvOptions } from '@launchql/env';
import { join } from 'path';
import { teardownPgPools } from 'pg-cache';
import { getPgEnvOptions } from 'pg-env';

import { LaunchQLProject } from '../src/core/class/launchql';
import { MigrateTestFixture, TestDatabase } from './index';
import { TestFixture } from './TestFixture';

export class CoreDeployTestFixture extends TestFixture {
  private databases: TestDatabase[] = [];
  private migrateFixture?: MigrateTestFixture;

  constructor(...fixturePath: string[]) {
    super(...fixturePath);
    this.migrateFixture = new MigrateTestFixture();
  }

  async setupTestDatabase(): Promise<TestDatabase> {
    if (!this.migrateFixture) {
      throw new Error('MigrateTestFixture not initialized');
    }
    
    const db = await this.migrateFixture.setupTestDatabase();
    this.databases.push(db);
    return db;
  }

  async deployModule(projectName: string, database: string, fixturePath: string[], logOnly: boolean = false): Promise<void> {
    const basePath = this.tempFixtureDir;
    const originalCwd = process.cwd();
    
    try {
      process.chdir(basePath);
      
      const project = new LaunchQLProject(basePath);
      
      const opts = getEnvOptions({ 
        pg: getPgEnvOptions({ database }),
        deployment: {
          fast: false,
          usePlan: true,
          logOnly
        }
      });

      await project.deploy(opts, projectName, undefined, true);
    } finally {
      process.chdir(originalCwd);
    }
  }

  async revertModule(projectName: string, database: string, fixturePath: string[], toChange?: string): Promise<void> {
    const basePath = this.tempFixtureDir;
    const originalCwd = process.cwd();
    
    try {
      const projectPath = join(basePath, 'packages', projectName);
      process.chdir(projectPath);
      
      const project = new LaunchQLProject(projectPath);
      
      const opts = getEnvOptions({ 
        pg: getPgEnvOptions({ database })
      });

      await project.revert(opts, projectName, toChange, true);
    } finally {
      process.chdir(originalCwd);
    }
  }

  async verifyModule(projectName: string, database: string, fixturePath: string[], toChange?: string): Promise<void> {
    const basePath = this.tempFixtureDir;
    const originalCwd = process.cwd();
    
    try {
      process.chdir(basePath);
      
      const project = new LaunchQLProject(basePath);
      
      const opts = getEnvOptions({ 
        pg: getPgEnvOptions({ database })
      });

      await project.verify(opts, projectName, toChange, true);
    } finally {
      process.chdir(originalCwd);
    }
  }

  async cleanup(): Promise<void> {
    for (const db of this.databases) {
      await db.close();
    }
    this.databases = [];
    
    if (this.migrateFixture) {
      await this.migrateFixture.cleanup();
    }
    
    await teardownPgPools();
    
    // Small delay to ensure connections are fully closed
    await new Promise(resolve => setTimeout(resolve, 10));
    
    super.cleanup();
  }
}
