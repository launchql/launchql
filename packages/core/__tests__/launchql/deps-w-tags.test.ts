import { TestFixture } from '../../test-utils';

let fixture: TestFixture;

beforeAll(() => {
  fixture = new TestFixture('sqitch');
});

afterAll(() => {
  fixture.cleanup();
});

it('getModuleExtensions for my-first', () => {
  const project = fixture.getModuleProject(['simple-w-tags'], 'my-first');
  expect(project.getModuleExtensions()).toMatchSnapshot();
});

it('getModuleExtensions for my-second', () => {
  const project = fixture.getModuleProject(['simple-w-tags'], 'my-second');
  expect(project.getModuleExtensions()).toMatchSnapshot();
});

it('getModuleExtensions for my-third', () => {
  const project = fixture.getModuleProject(['simple-w-tags'], 'my-third');
  expect(project.getModuleExtensions()).toMatchSnapshot();
});

it('getModuleDependencies for my-third', () => {
  const project = fixture.getModuleProject(['simple-w-tags'], 'my-third');

//   console.log(project.getAvailableModules());
//   console.log(project.getModuleMap());
//   console.log(project.getModuleExtensions());

  expect(project.getModuleDependencies('my-third')).toMatchSnapshot();
});
