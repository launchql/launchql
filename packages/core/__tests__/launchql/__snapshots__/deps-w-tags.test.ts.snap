// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getModuleDependencies for my-third 1`] = `
{
  "modules": [
    "my-second",
  ],
  "native": [
    "citext",
    "plpgsql",
    "pgcrypto",
  ],
}
`;

exports[`getModuleExtensions for my-first 1`] = `
{
  "external": [
    "citext",
    "plpgsql",
    "pgcrypto",
  ],
  "resolved": [
    "citext",
    "plpgsql",
    "pgcrypto",
    "my-first",
  ],
}
`;

exports[`getModuleExtensions for my-second 1`] = `
{
  "external": [
    "citext",
    "plpgsql",
    "pgcrypto",
  ],
  "resolved": [
    "citext",
    "plpgsql",
    "pgcrypto",
    "my-first",
    "my-second",
  ],
}
`;

exports[`getModuleExtensions for my-third 1`] = `
{
  "external": [
    "citext",
    "plpgsql",
    "pgcrypto",
  ],
  "resolved": [
    "citext",
    "plpgsql",
    "pgcrypto",
    "my-first",
    "my-second",
    "my-third",
  ],
}
`;
